packages/ic-certificate-verification/src/signature_verification/mod.rs

```diff
@@ -0,0 +1,32 @@
+use self::signature_cache::{SignatureCache, SignatureCacheEntry};
+use crate::CertificateVerificationError;
+use miracl_core_bls12381::bls12381::bls::{core_verify, BLS_OK};
+
+mod signature_cache;
+
+#[cfg(test)]
+mod reproducible_rng;
+
+#[cfg(test)]
+mod tests;
+
+pub fn verify_signature(
+    pk: &[u8],
+    sig: &[u8],
+    msg: &[u8],
+) -> Result<(), CertificateVerificationError> {
+    let entry = SignatureCacheEntry::new(pk, sig, msg);
+
+    if SignatureCache::global().contains(&entry) {
+        return Ok(());
```

Comment:
Note: it could be useful to somehow expose the cache hits/misses for metrics to see how many verifications are costly.

Reply:
Thanks, yes I discussed this with @r-birkner and we agreed we wanted to take the shorter path and get this released ASAP. There's a redesign planned for ICX Proxy and that will be a good chance to rethink how we are handling metrics.